File: 序号/633. 平方数之和.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        for i in range(int(c**0.5)+1): # c 为i j的平方和，i大小不能超过根号c
            j=int((c-i*i)**0.5)
            if i*i+j*j==c:
                return True
        return False

File: 序号/633. 平方数之和.kt
class Solution {
  val limit = sqrt( Int.MAX_VALUE.toDouble()).toInt()
  val arr  = mutableSetOf<Int>().apply { 
      for (i in 0..limit) add(i*i)
  }

  fun judgeSquareSum(c: Int): Boolean {
      for (i in 0..limit){
          if (arr.contains(c-i*i))return true
      }
      return false
  }
}